src/main/java/teammates/ui/output/FeedbackSessionData.java

```diff
@@ -443,14 +462,17 @@ private void hideInformationForStudentAndInstructor() {
         setPublishedEmailEnabled(null);
         setGracePeriod(null);
         setCreatedAtTimestamp(0);
+    }
 
+    private void hideInformationForStudentAndInstructor(List<DeadlineExtension> deadlineExtensions, String userEmail) {
```

Comment:
hideInformationForXXX is not the right place to set deadlineExtensions either. Expected behaviour of the method is to hide existing information not set values.

Let's follow what has been done already. Set all deadlines in the constructor, like how it was done before the change. Then use hideXX methods to filter out data that shouldn't be there.

For full detail, no need to filter out any deadlines.

For instructor submission/result, student deadlines should be empty. instructor deadlines should only have the particular instructors deadline.

For student submission/result, similar to instructor above.

Reply:
Hi @samuelfangjw, how about this earlier comment then? In the constructor, all the deadlines are set without filtering. 

Only when the hide commands are invoked, then the filtering is done. This was implemented to fix the earlier review comment. 

> Concerned about the potential implications of doing the filtering here. Idea here is to hide other user deadlines from the student. Now that we have two separate constructors, there's always a chance that someone uses the more general constructor (avoiding the filtering altogether), because one would expect this filtering to be have been done in hideInformationForStudent.
> 
> Might be better to do this filtering in hideInformationFromStudent/Instructor instead.

Reply:
Yes and they are filtered out when hideXX is called. No conflict here.

Reply:
Original issue is someone calling using the more general constructor (without filtering deadlines in the original implementation), then calling hide expecting that sensitive information is filtered out. Based on the expected behaviours of these methods, one would expect any sensitive information to be hidden but that is not the case in this scenario.

Reply:
ok fixed

Reply:
Sorry didn't see it being set in the constructor. Issue is that it's being set again in hide, which is not expected. 
Can do a filter on existing deadline extensions in `hideInformationForInstructor` and `hideInformationForStudent`, no need to set this again.

Reply:
@samuelfangjw, yep, its fixed, the changes should be good and much cleaner now